File: Part8/Go/FunctionConcepts/FunctionConcepts_12.go
// Computing factorial with loop
package main

import "fmt"

func factorial ( FactorialNumber_Par int ) int {
    totalFactorial := 1
    
    for factorialNumber := 1; factorialNumber <= FactorialNumber_Par; factorialNumber++ {
        totalFactorial = totalFactorial * factorialNumber
    }
    
    return totalFactorial
}

func main() {
    factorialResult := factorial ( 10 )
    fmt.Printf ( "Factorial of 10 is: %d\n", factorialResult )
}

File: Part8/C#/FunctionConcepts/FunctionConcepts_12.cs
// Computing factorial with loop
using System;

class Program
{
    static int Factorial ( int FactorialNumber_Par )
    {
        int totalFactorial = 1;
        
        for ( int factorialNumber = 1; factorialNumber <= FactorialNumber_Par; factorialNumber++ )
        {
            totalFactorial = totalFactorial * factorialNumber;
        }
        
        return totalFactorial;
    }

    static void Main ()
    {
        int factorialResult = Factorial ( FactorialNumber_Par: 10 );
        Console.WriteLine ( "Factorial of 10 is: {0}", factorialResult );
    }
}
